### PR TITLE
README added for molecular subtype integrate module

### PR DESCRIPTION
**Pull request by runjin326**
#### What scientific question is your analysis addressing?
Modified README to include explanation for cancer group assignment.

#### What was your approach?
First showed which cancer groups would just use `harmonized_diagnosis`.
Then showed which cancer groups would be `harmonized_diagnosis` minus the subtype information.
Lastly, showed a table indicating additional modifications to the cancer groups for formatting/simplification reasons.

#### What GitHub issue does your pull request address?
https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1246

#### Which areas should receive a particularly close look?
Please check to see whether the description is clear enough. 
We could also directly show a table containing how each `harmonized_diagnosis` is mapped to `cancer_group`.

#### Is there anything that you want to discuss further?
See above.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes.

#### What types of results are included (e.g., table, figure)?
README file with description and tables.

#### What is your summary of the results?
NA

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by sjspielman** [comment]
Thanks for this @runjin326 ! Going through it now.

I want to note here that this module's notebook itself seems to be missing some important text - line 29 ends in a sentence fragment. 
```
If samples are not processed by a molecular-subtyping-* module then the br
```

This needs to be fleshed out to enable clear, unambiguous descriptions of the groupings. CC @jharenza @kgaonkar6 , maybe we need a separate issue/PR for this?

**Comment by runjin326** [comment]
@sjspielman - thanks so much for the review. I have now added a separate Rmd showing how the tables in README are generated. (I feel like a separate Rmd is clearer than additional couple lines at the end of 01 script)

I also modified the language in the README a little bit to make it more clear how those lists/tables were generated. Let me know if they all make sense.

**Comment by sjspielman** [comment]
Hi @runjin326 , adding that separate file is probably the right way to go here. However, I have to say, I am really confused by what this file is and isn't doing. I'm going to loop @jharenza in because I think I must be just missing something here. Possibly what's missing is there is no clear definition of the procedure for cancer groupings in the first place so it's challenging to assess how these files bring that information out. I might suggest bringing in someone with more history with this module to add in some clearer explanations in general?

Either way, my sense was - the goal is here is to precisely show how the cancer groups were derived so it is clear how more granular diagnoses relate to the final cancer groupings. So, I would expect this file to clearly show me which samples or other groupings are coalescing into cancer groups, of which there are 59 (and an `NA`) in the histologies file. If this is not the goal, I have misunderstood.

**Comment by jharenza** [comment]
Hi @sjspielman. @runjin326 and I just chatted offline about this and she is going to make some changes because I was also a little confused :). The plan is to create a final table which has all harmonized diagnoses mapped to their new cancer groups and to describe in text what was done to get there. In brief, it was mostly removal of the molecular subtype (if exists) and then collapsing into manual groups which I chose based on what made the most sense for plotting/analysis. There were a few aesthetic/nomenclature changes, which were removals of `/` and `;` and all abbreviations in parentheses to collapse groups/make plotting easier, so we will add that into a paragraph to describe.

**Comment by jaclyn-taroni** [comment]
I filed a PR to this branch: https://github.com/runjin326/OpenPBTA-analysis/pull/1  because I thought having some code to talk about would be helpful when I describe some ideas I had. More info over there, but to summarize my concerns: 

* We probably want to create tables from the `01` output rather than what's in the data release. There are at least two ways to accomplish that.
* I think it would be helpful to have all of the info in `has_subtype_removed` and `additional_modi` in a table that anyone can inspect.
* It'd be great to save a table specifically for what needs to go into the README. I _think_ that's what I accomplished.

File: analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd [hunk 1]
```diff
@@ -203,6 +203,59 @@ histology<- histology %>%
 
 ```
 
+#### Write out table for `harmonized_diagnosis` -> `cancer_group` mapping
+
+To make it easy for people to examine the mapping between these two fields, we'll write out a table to `results/`.
```
Comment (sjspielman) [inline analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:208]:
```suggestion
To make it easier for people to examine the mapping between these two fields, we'll write out a table to `results/`.
```
:wink:

Comment (sjspielman) [reply analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:208]: (either way is fine, feel free to ignore)

File: analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd [hunk 2]
```diff
@@ -203,6 +203,59 @@ histology<- histology %>%
...
+# modifications beyond that have been performed.
+mapping_table <- histology %>% 
+  # Consider removing this sample? -- seems to be a special case, see chunk above
+  # dplyr::filter(sample_id != "7316-2577") %>%
+  dplyr::select(harmonized_diagnosis, 
+                cancer_group) %>%
+  dplyr::distinct() %>%
+  dplyr::filter(!is.na(harmonized_diagnosis)) %>%
+  # This next step removes the subtypes from harmonized diagnosis
+  dplyr::mutate(cancer_group_pre_mod = str_extract(harmonized_diagnosis,
```
Comment (sjspielman) [inline analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:224]: I would rename this to `harmonized_diagnosis_nosubtype` or similar. I find this column name `cancer_group_pre_mod` confusing without a lot of context.
Edit - I see later it is renamed to `stripped_subtype`. I might just start with this name.

File: analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd [hunk 3]
```diff
@@ -251,3 +304,10 @@ histology %>%
   arrange(Kids_First_Biospecimen_ID) %>%
   write_tsv("results/pbta-histologies.tsv")
 ```
+
+### Session Info
```
Comment (sjspielman) [inline analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:308]: 👍

**Review by sjspielman — APPROVED** [review]
This is clearer, thanks @jaclyn-taroni for going through it! I made 1-2 quick comments to look at before merge.

**Comment by sjspielman** [comment]
Noting that in interest of time I made the requested changes. This is now ready for merge.

**Pull request merged**